pkg/run/state/state.go

```diff
@@ -15,6 +16,64 @@ type State struct {
 	kubeletProbeFailures  int
 	kubeletAPIPodRunning  int
 	kubeletLogsPodRunning int
+
+	promVersion prometheus.Gauge
+
+	promStateReady            prometheus.Gauge
+	promKubeletAPIPodRunning  prometheus.Gauge
+	promKubeletLogsPodRunning prometheus.Gauge
+
+	promKubeletProbeFailures prometheus.Counter
+}
+
+// NewState instantiate a state with the associated prometheus metrics
+func NewState() (*State, error) {
+	s := &State{
+		promVersion: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name:        "pupernetes_version",
+			Help:        "Pupernetes version",
+			ConstLabels: prometheus.Labels{},
+			// TODO record all versions in labels. hyperkube: "1.10.1", etcd: "3.11.1", ...
+		}),
+		promStateReady: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "pupernetes_ready",
+			Help: "Boolean for pupernetes readiness",
+		}),
+		promKubeletAPIPodRunning: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "pupernetes_kubelet_api_pods_running",
+			Help: "Number of kubelet API pods running",
+		}),
+		promKubeletLogsPodRunning: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "pupernetes_kubelet_logs_pods_running",
+			Help: "Number of kubelet logs pods running",
+		}),
+		promKubeletProbeFailures: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: "pupernetes_kubelet_probe_failures",
+			Help: "Total number of kubelet probe failures",
+		}),
+	}
+	err := prometheus.Register(s.promVersion)
```

Comment:
you could remove the `error` from `NewState` and use `MustRegister` instead of `Register` which would make the program panic if failing, this would clean up all the error handling code as well

Reply:
@mfpierre I wasn't sure about that, do you think it worth it ?
Why panic should be better than an error ?

Reply:
Don't you exit anyway if it fails?